source/ed/ed_reader.cpp

```diff
@@ -619,6 +622,138 @@ void EdReader::fill_stop_points(nt::Data& data, pqxx::work& work) {
         }
         data.pt_data->stop_points.push_back(sp);
         this->stop_point_map[const_it["id"].as<idx_t>()] = sp;
+        this->uri_to_idx_stop_point[const_it["uri"].as<std::string>()] = const_it["id"].as<idx_t>();
+    }
+}
+
+void EdReader::fill_access_point_field(navitia::type::AccessPoint* access_point,
+                                       const pqxx::result::iterator const_it,
+                                       const bool from_access_point,
+                                       const std::string& sp_id) {
+    if (!const_it["pathway_mode"].is_null()) {
+        access_point->pathway_mode = const_it["pathway_mode"].as<unsigned int>();
+    }
+    if (!const_it["is_bidirectional"].is_null()) {
+        const bool is_bidirectional = const_it["is_bidirectional"].as<bool>();
+        if (is_bidirectional) {
+            access_point->is_entrance = true;
+            access_point->is_exit = true;
+        } else {
+            if (from_access_point) {
+                access_point->is_entrance = true;
+                access_point->is_exit = false;
+            } else {
+                access_point->is_entrance = false;
+                access_point->is_exit = true;
+            }
+        }
```

Comment:
maybe we can reduced nested `if` a little bit?
```suggestion
       [access_point](){
           if (is_bidirectional) {
               access_point->is_entrance = true;
               access_point->is_exit = true;
               return;
           }          
           if (from_access_point) {
               access_point->is_entrance = true;
               access_point->is_exit = false;
               return;
           }
           access_point->is_entrance = false;
           access_point->is_exit = true;
       }();
        
```